src/Components/Analyzers/src/ComponentParameterUsageAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+        {
+            SupportedDiagnostics = ImmutableArray.Create(new[]
+            {
+                DiagnosticDescriptors.ComponentParametersShouldNotBeSetOutsideOfComponents,
+            });
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!ComponentSymbols.TryCreate(context.Compilation, out var symbols))
+                {
+                    // Types we need are not defined.
+                    return;
+                }
+
+                context.RegisterOperationBlockStartAction(startBlockContext =>
+                {
+                    var assignedPropertyReferences = new List<IPropertyReferenceOperation>();
+                    startBlockContext.RegisterOperationAction(context =>
+                    {
+                        foreach (var operation in context.Operation.Children)
+                        {
+                            if (operation.Kind == OperationKind.PropertyReference)
+                            {
+                                assignedPropertyReferences.Add((IPropertyReferenceOperation)operation);
+                            }
+                        }
+                    }, OperationKind.SimpleAssignment, OperationKind.CompoundAssignment, OperationKind.CoalesceAssignment);
+
+                    startBlockContext.RegisterOperationBlockEndAction(context =>
+                    {
+                        for (var i = 0; i < assignedPropertyReferences.Count; i++)
+                        {
+                            var propertyReference = assignedPropertyReferences[i];
+                            var componentProperty = propertyReference.Member;
+                            var propertyAttributes = componentProperty.GetAttributes();
+
+                            if (!ComponentFacts.HasParameterAttribute(symbols, propertyAttributes))
+                            {
+                                // This is not a property reference that we care about, it is not decorated with [Parameter].
+                                continue;
+                            }
+
+                            var propertyLocationContainingType = componentProperty.ContainingType;
+                            if (!ComponentFacts.IsComponent(symbols, propertyLocationContainingType))
+                            {
+                                // Someone referenced a property as [Parameter] inside something that is not a component.
+                                continue;
+                            }
+
+                            var assignmentLocationContainingType = startBlockContext.OwningSymbol?.ContainingType;
```

Comment:
Is the reason for the start/end that you need the `owningSymbol`?

Reply:
I need to get the containing type that holds the assignment operation. Is there another way to get that info other than the owning blocks symbol?